libraries/common/test.effekt

```diff
@@ -87,4 +88,11 @@ def suite(name: String) { body: => Unit / Test }: Bool = {
     println(ANSI::GREEN ++ "All tests passed (" ++ passed.show ++ " passed)" ++ ANSI::RESET)
     true
   }
+}
+
+/// Use as `def main() = mainSuite("...") { ... }`
+def mainSuite(name: String) { body: => Unit / Test }: Unit = {
```

Comment:
I have no idea what the name of this function should be, but the goal is to use it in automated tests / in standalone test binaries, for example like:
```scala
def main() = mainSuite("My Test Suite") {
  test("Addition works") {
    assertEqual(2 + 2, 4)
  }
}
```

Reply:
Also, should this return `Unit` or `Nothing`?

Reply:
I would go with Unit considering that main currently needs to return Unit

Reply:
Another thought related to the design of `suite` and tests: would it make sense to first create a reified representation of the test results that can be queried to then render / print it / translate to exit code?

Reply:
Sure, the only problem here is that in the ideal case, what I call `mainSuite` could also be able to install CLI args in order to filter tests / run only a single one.

But yes, the `suite` itself could just return reified results, then we'd have something else that would translate them.

Reply:
The output could / should even be "streaming" using effects :)

Reply:
I'm fine with doing that, but I would prefer to do that part in #542 instead of here.